ksp_plugin/vessel.cpp

```diff
@@ -246,13 +247,14 @@ void Vessel::AdvanceTime() {
   AppendToVesselTrajectory(&Part::psychohistory_begin,
                            &Part::psychohistory_end,
                            *psychohistory_);
-  {
-    absl::MutexLock l(&prognosticator_lock_);
-    if (prognostication_ == nullptr) {
-      AttachPrediction(std::move(prediction));
-    } else {
-      AttachPrediction(std::move(prognostication_));
-    }
+
+  // Attach the prognostication, if there is one.  Otherwise fall back to the
+  // prediction.
```

Comment:
`[...] to the pre-existing prediction`. Either way we are attaching something which becomes « the prediction ».